content/docs/guides/go-application.md

```diff
@@ -0,0 +1,123 @@
+go get github.com/prometheus/client_golang/prometheus/promauto
+go get github.com/prometheus/client_golang/prometheus/promhttp
+```
+
+## How Go exposition works
+
+To expose Prometheus metrics in a Go application, you need to provide a `/metrics` HTTP endpoint. You can use the [`prometheus/promhttp`](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp) library's HTTP [`Handler`](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#Handler) as the handler function.
+
+This minimal application, for example, would expose the default metrics for Go applications via `localhost:2112/metrics`:
+
+```go
+package main
+
+import (
+        "net/http"
+
+	      "github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func main() {
+        http.Handle("/metrics", promhttp.Handler())
+        http.ListenAndServe(":2112", nil)
+}
+```
+
+To start the application:
+
+```bash
+go run main.go
+```
+
+To access the metrics:
+
+```bash
+curl http://localhost:2112/metrics
+```
+
+## Adding your own metrics
+
+The application [above](#how-go-exposition-works) exposes only the default Go metrics. You can also register your own custom application-specific metrics. This example application exposes a `myapp_queued_ops` [gauge](/docs/concepts/metric_types/#gauge) that registers the number of currently queued operations. Every 2 seconds, the gauge is incremented by one.
+
+```go
+package main
+
+import (
+        "net/http"
+        "time"
+
+        "github.com/prometheus/client_golang/prometheus"
+        "github.com/prometheus/client_golang/prometheus/promauto"
+        "github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func recordMetrics() {
+        go func() {
+                for {
+                        opsQueued.Add(1)
+                        time.Sleep(2 * time.Second)
+                }
+        }()
+}
+
+var (
+        opsQueued = promauto.NewGauge(prometheus.GaugeOpts{
+                Name: "myapp_queued_ops",
+                Help: "The number of operations currently queued",
+        })
+)
+
+func main() {
+        recordMetrics()
+
+        http.Handle("/metrics", promhttp.Handler())
+        http.ListenAndServe(":2112", nil)
+}
+```
+
+To run the application:
+
+```bash
+go run main.go
+```
+
+To access the metrics:
+
+```bash
+curl http://localhost:2112/metrics
+```
+
+In the metrics output, you'll see the help text, type information, and current value of the `myapp_queued_ops` gauge:
+
+```
+# HELP myapp_queued_ops The number of operations currently queued
+# TYPE myapp_queued_ops gauge
+myapp_queued_ops 1
+```
+
+You can [configure](/docs/prometheus/latest/configuration/configuration/#<scrape_config>) a locally running Prometheus instance to scrape metrics from the application. Here's an example `prometheus.yml` configuration:
+
+```yaml
+scrape_configs:
+- job_name: myapp
+  scrape_interval: 10s
+  static_configs:
+  - targets:
+    - localhost:2112
+```
```

Comment:
Maybe add a blurb at the end mentioning that there are different metric types and more advanced stuff (like non-global registries), with some pointers?